### PR TITLE
Command line interface

### PR DESCRIPTION
**Pull request by lemmoi**
Adds a command line interface, where all inputs are specified in a `.yml` file passed to the executable `aimless_driver`. An example is provided in the `examples` directory with lots of comments for documentation. I think all the input in this file is fully validated. 

There are two new tests for this 
- `test_parsing.py` mostly verfies that the inputs are validated and checks that bad inputs are rejected
- `test_end2end.py` is a full integration test based on the input file for our sample 3 particle system. Runs aimless shooting, colvar calculation, and likelihood maximization all in a row from scratch.

Another important change - in `main.py` we were turning off all logging because CP2K constantly complains that file names are being truncated (even though it doesn't seem to actually do anything?). I've just manually filtered those out now so that logging can be turned on in prep for better logging.

### CODE REVIEW

File: transition_sampling/driver.py [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+import yaml
+import sys
+
+from schema import Schema, And, Optional, Use, Or
+
+master_schema = Schema({Optional("md_inputs"): dict,
+                        Optional("colvar_inputs"): dict,
+                        Optional("likelihood_inputs"): dict})
+
+name_schema = Schema({"md_inputs": {"aimless_inputs": {"output_name": str}}}, ignore_extra_keys=True)
```
Comment (lemmoi) [inline transition_sampling/driver.py:18]: Remove

File: transition_sampling/driver.py [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+
+    likelihood_schema = Schema({Optional("max_cvs"): Or(None,
+                                                        And(int, lambda x: x >= 1,
+                                                            error="max_cvs must be null or >= 1")),
+                                "output_name": str,
+                                Optional("csv_input"): Or(None, And(str, Use(check_is_file))),
+                                Optional("colvar_input"): Or(None, And(str, Use(check_is_file))),
+                                Optional("n_iter"): And(int, lambda x: x >= 1,
+                                                        error="n_iter must be >= 1"),
+                                Optional("use_jac"): bool})
```
Comment (lemmoi) [inline transition_sampling/driver.py:233]: Set defaults

File: transition_sampling/tests/driver_tests/test_parsing.py [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+            self.inputs[field] = None
+            with self.assertRaises((SchemaError, SystemExit),
+                                   msg=f"{field} should not be allowed to be None without earlier parsing"):
+                driver.parse_colvar(self.inputs)
+
+            # reset input to check the others
+            self.inputs[field] = original_value
+
+    def test_parsing_earlier_fields(self):
+        _clear_globals()
```
Comment (lemmoi) [inline transition_sampling/tests/driver_tests/test_parsing.py:175]: Remove

**Review by ldgibson — APPROVED** [review]
LGTM

**Pull request merged**